core/store/src/trie/update.rs

```diff
@@ -226,11 +264,23 @@ impl<'a> TrieUpdateIterator<'a> {
             None => None,
         };
         trie_iter.seek(&start_offset)?;
-        let committed_iter = state_update.committed.range(start_offset.clone()..);
+        let committed_iter =
+            state_update.committed.range(start_offset.clone()..).map(|(k, changes)| {
+                (
+                    k,
+                    &changes
+                        .last()
+                        .as_ref()
+                        .expect("Committed entry should have at least one change.")
+                        .1,
+                )
+            });
         let prospective_iter = state_update.prospective.range(start_offset..);
-        let overlay_iter =
-            MergeIter { left: committed_iter.peekable(), right: prospective_iter.peekable() }
-                .peekable();
+        let overlay_iter = MergeIter {
+            left: (Box::new(committed_iter) as Box<dyn Iterator<Item = _>>).peekable(),
+            right: (Box::new(prospective_iter) as Box<dyn Iterator<Item = _>>).peekable(),
```

Comment:
I feel like there is something wrong and I should not be this explicit, but compiler forces me to do this.

Reply:
I think it is because of the box. `Box::new(committed_iter)` does not have virtual table while `Box<dyn Iterator<Item = _>>` needs one so probably compiler wants you to be explicit about it since it creates a virtual table.